torchaudio/csrc/sox/effects_chain.cpp

```diff
@@ -69,49 +69,46 @@ int tensor_input_drain(sox_effect_t* effp, sox_sample_t* obuf, size_t* osamp) {
   *osamp -= *osamp % num_channels;
 
   // Slice the input Tensor
-  const auto tensor_ = [&]() {
+  auto chunk = [&]() {
     auto i_frame = index / num_channels;
     auto num_frames = *osamp / num_channels;
     auto t = (priv->channels_first)
         ? tensor.index({Slice(), Slice(i_frame, i_frame + num_frames)}).t()
         : tensor.index({Slice(i_frame, i_frame + num_frames), Slice()});
-    return t.reshape({-1}).contiguous();
+    return t.reshape({-1});
   }();
 
-  // Convert to sox_sample_t (int32_t) and write to buffer
-  SOX_SAMPLE_LOCALS;
-  switch (tensor_.dtype().toScalarType()) {
+  // Convert to sox_sample_t (int32_t)
+  switch (chunk.dtype().toScalarType()) {
     case c10::ScalarType::Float: {
-      auto ptr = tensor_.data_ptr<float_t>();
-      for (size_t i = 0; i < *osamp; ++i) {
-        obuf[i] = SOX_FLOAT_32BIT_TO_SAMPLE(ptr[i], effp->clips);
-      }
+      // Need to convert to 64-bit precision so that
+      // values around INT32_MIN/MAX are handled correctly.
+      chunk = chunk.to(c10::ScalarType::Double);
+      chunk *= 2147483648.;
+      chunk.clamp_(INT32_MIN, INT32_MAX);
+      chunk = chunk.to(c10::ScalarType::Int);
```

Comment:
I'm not sure that this will guarantee that chunk is contiguous afterwards. I'd call into "contiguous" anyway to be sure. Same applies for the other cases.

Reply:
cool. will update